compiler/src/dotty/tools/io/JarFS.scala

```diff
@@ -0,0 +1,28 @@
+package dotty.tools
+package io
+
+import java.nio.file.{FileSystem, FileSystems}
+import java.nio.file.Files
+
+import java.util.jar._
+
+import scala.collection.JavaConverters._
+
+class JarFS private (private[this] var jarFS: FileSystem) {
+
+  def getRoot(): AbstractFile = {
+    val root = jarFS.getRootDirectories.iterator().next()
+    new PlainDirectory(Directory(root))
+  }
+
+  def close() = jarFS.close()
+}
+
+object JarFS {
+  def create(path: Path): JarFS = {
+    assert(path.extension == "jar")
+    val env = Map("create" -> "true").asJava
```

Comment:
@allanrenucci here is the documentation:
https://docs.oracle.com/javase/7/docs/technotes/guides/io/fsp/zipfilesystemprovider.html
https://docs.oracle.com/javase/7/docs/technotes/guides/io/fsp/zipfilesystemproviderprops.html

Reply:
I would add these links in comments if they're useful to understand the code